source/lac/scalapack.cc

```diff
@@ -1320,9 +1506,72 @@ void ScaLAPACKMatrix<NumberType>::load_parallel(const char *filename)
 
 
 
+namespace internal
+{
+  namespace
+  {
+
+    template <typename NumberType>
+    void scale_columns(ScaLAPACKMatrix<NumberType>       &matrix,
+                       const ArrayView<const NumberType> &factors,
+                       const bool                         grid_mpi_process_is_active)
+    {
+      Assert(matrix.n()==factors.size(),ExcDimensionMismatch(matrix.n(),factors.size()));
+
+      if (grid_mpi_process_is_active)
+        for (unsigned int i=0; i<matrix.local_n(); ++i)
+          {
+            const NumberType s = factors[matrix.global_column(i)];
+
+            for (unsigned int j=0; j<matrix.local_m(); ++j)
+              matrix.local_el(j,i) *= s;
+          }
+    }
+
+    template <typename NumberType>
+    void scale_rows(ScaLAPACKMatrix<NumberType>       &matrix,
+                    const ArrayView<const NumberType> &factors,
+                    const bool                         grid_mpi_process_is_active)
+    {
+      Assert(matrix.m()==factors.size(),ExcDimensionMismatch(matrix.m(),factors.size()));
+
+      if (grid_mpi_process_is_active)
+        for (unsigned int i=0; i<matrix.local_m(); ++i)
+          {
+            const NumberType s = factors[matrix.global_row(i)];
+
+            for (unsigned int j=0; j<matrix.local_n(); ++j)
+              matrix.local_el(i,j) *= s;
+          }
+    }
+
+  }
+}
+
+
+
+template <typename NumberType>
+template <class InputVector>
+void ScaLAPACKMatrix<NumberType>::scale_columns(const InputVector &factors)
+{
+  internal::scale_columns(*this, make_array_view(factors),
+                          this->grid->mpi_process_is_active);
```

Comment:
minor nitpick, i think it's easier to read as
```
  if (this->grid->mpi_process_is_active)
    internal::scale_columns(...);
```
that is, when you move that `if` outside of internal function which actually does the job.